app/controllers/brapi/v1/germplasm_controller.rb

```diff
@@ -0,0 +1,250 @@
+    LEFT JOIN countries AS countries_registered_from_accessions ON plant_variety_country_registered_from_accessions.country_id = countries_registered_from_accessions.id 
+    LEFT JOIN taxonomy_terms ON plant_lines.taxonomy_term_id = taxonomy_terms.id
+    "
+    
+    
+    # order
+    order_query =  " ORDER BY plant_accessions.id desc"
+    
+    if count_mode
+      total_query = "SELECT COUNT(*) FROM ("+select_query + joins_query + where_query + order_query+") AS total_entries_count"
+
+      puts total_query
+      result_object = execute_statement(total_query, where_atts)
+    else
+      # pagination
+      page = get_page
+      pageSize = get_pageSize
+      pagination_query = pagination_query(page, pageSize)
+      
+      total_query = select_query + joins_query + where_query + order_query + pagination_query
+      
+      json_wrapping_query = "SELECT row_to_json(row) from ("+total_query+") row"
+      
+      result_object = execute_statement(json_wrapping_query, where_atts)
+    end
+    
+    records = result_object.values
+    result_object
+  end
+
+
+  def pagination_query(page, pageSize)
+    return " LIMIT "+pageSize.to_s+" OFFSET "+((page-1)*pageSize).to_s  
+  end
+  
+  
+  def get_page
+    return (params[:page].to_i != 0)?params[:page].to_i : 1
+  end
+  
+  def get_pageSize
+    return (params[:pageSize].to_i != 0)?params[:pageSize].to_i : 1000 
+  end
+  
+  def json_metadata(pageSize, currentPage, totalCount, totalPages)
+    json_metadata = {
+      status: [],
+      files: [],
+      pagination: {
+        pageSize: pageSize, # old 'per_page'
+        currentPage:  currentPage, # old 'page'
+        totalCount: totalCount, # old 'total_count'
+        totalPages: totalPages 
+      }
+    }
+  end
+  
+
+
+  def authenticate_api_key!
+    unless api_key_token.present?
+      render json: '{"reason": "BIP API requires API key authentication"}', status: 401
+      return
+    end
+    unless api_key.present?
+      if api_key_token == I18n.t('api.general.demo_key')
+        render json: '{"reason": "Please use your own, personal API key"}', status: 401
+      else
+        render json: '{"reason": "Invalid API key"}', status: 401
+      end
+    end
+  end
+
+  def api_key_token
+    return @api_key_token if defined?(@api_key_token)
+    token = params[:api_key] || request.headers["X-BIP-Api-Key"]
+    @api_key_token = ApiKey.normalize_token(token)
+  end
+
+  def api_key
+    return @api_key if defined?(@api_key)
+    @api_key = api_key_token && ApiKey.find_by(token: api_key_token)
+  end
+
+
+  
+  def execute_statement(sql, atts)
+      connection = ActiveRecord::Base.connection.raw_connection
+
+      connection.prepare('brapi_statement', sql)
+      
+      if( (atts != nil) && !(atts.empty?) )
+        results = connection.exec_prepared("brapi_statement", atts)
```

Comment:
OK, I see you pass the user-provided attribute values to `exec_prepared` so I think this is safe, code injection wise.